rx-java/src/main/resources/vertx-rxjava/template/common.templ

```diff
@@ -238,6 +238,15 @@
       return deflt;
     }
   }
+
+  def hasToString(methods) {
```

Comment:
have you checked with method inheritance ?

Reply:
You mean if someone defines `toString` in a `concrete=false` Vert.x object but not in the child?

Reply:
@vietj just added a commit with test files to make sure they can get compiled

Reply:
actually I'm simply wondering why aren't we just delegating always toString() like equals and hashcode and why are we trying to be smart here ?

Reply:
Because a user may want to declare toString in a Vert.x object interface if it has a special meaning. We do this with Buffer.

Reply:
I believe we should always generate toString() because the implementation may implement a toString() method that the interface does not declare, that cannot hurt is would lead to better overall usability of RX generated API

Reply:
for instance `io.vertx.core.impl.FutureImpl` overrides `toString` with:

```
  public String toString() {
    synchronized (this) {
      if (succeeded) {
        return "Future{result=" + result + "}";
      }
      if (failed) {
        return "Future{cause=" + throwable.getMessage() + "}";
      }
      return "Future{unresolved}";
    }
  }
```

and `io.vertx.core.Future` does redeclare `toString()`

Reply:
There's a misunderstanding. The `toString` method is always generated. It is generated like a normal method if it's declared in the interface, or as an extra method otherwise.

Reply:
`SocketAddressImpl` is similar to `FutureImpl` in this regard, and there's a test for it:
https://github.com/vert-x3/vertx-rx/pull/107/files#diff-374a9f7b30e06a9f221f8451593e1f19R38

Reply:
ah ok, great :-)